iib/workers/tasks/utils.py

```diff
@@ -1293,3 +1293,41 @@ def get_bundle_metadata(
         for pullspec in operator_csv.get_pullspecs():
             bundle_metadata['found_pullspecs'].add(pullspec)
     return bundle_metadata
+
+
+def filter_deprecation_operator_from_db(
+    deprecation_bundles: List[str], index_db_file: str
+) -> List[str]:
+    """
+    Check if operator marked for deprecation is available in operator db, if not drop them.
+
+    :param List[str] deprecation_bundles: List of bundles to be depricated
+    :param str index_db_file: Path of operator database
+
+    :return: A List of operator bundle to be deprecated
+    :raises:
+    """
+    try:
+        with sqlite3.connect(index_db_file) as conn:
```

Comment:
By this line you are introducing new dependency - `sqlite3`.
I think, we should use `opm` to get relevant data from index image DB.
You can use `get_list_bundles` function - you do no need to implement anything new.

Please rebase and check the code if it is possible to use `get_list_bundles`